java/src/jmri/managers/DefaultShutDownManager.java

```diff
@@ -303,6 +303,14 @@ private void doShutdown(int status, boolean exit) {
                     return;
                 }
             }
+            boolean abort = jmri.util.ThreadingUtil.runOnGUIwithReturn(() -> {
+                return jmri.configurexml.StoreAndCompare.checkPermissionToStoreIfNeeded();
+            });
+            if (abort) {
+                log.info("User aborted due to unsaved changes and has not permission to store");
```

Comment:
Alternate message:

User aborted the shutdown request due to not having permission to store changes

Reply:
Thanks. Fixed.